yasa/evaluation.py

```diff
@@ -0,0 +1,1501 @@
+        return summary
+
+
+################################################################################
+# SLEEP STATISTICS
+################################################################################
+
+
+class SleepStatsAgreement:
+    """
+    Evaluate agreement between sleep statistics reported by two different scorers.
+    Evaluation includes bias and limits of agreement (as well as both their confidence intervals),
+    various plotting options, and calibration functions for correcting biased values from the
+    observed scorer.
+
+    Features include:
+    * Get summary calculations of bias, limits of agreement, and their confidence intervals.
+    * Test statistical assumptions of bias, limits of agreement, and their confidence intervals,
+    and apply corrective procedures when the assumptions are not met.
+    * Get bias and limits of agreement in a string-formatted table.
+    * Calibrate new data to correct for biases in observed data.
+    * Return individual calibration functions.
+    * Visualize discrepancies for outlier inspection.
+    * Visualize Bland-Altman plots.
+
+    .. seealso:: :py:meth:`yasa.Hypnogram.sleep_statistics`
+
+    .. versionadded:: 0.7.0
+
+    Parameters
+    ----------
+    ref_data : :py:class:`pandas.DataFrame`
+        A :py:class:`pandas.DataFrame` with sleep statistics from the reference scorer.
+        Rows are unique observations and columns are unique sleep statistics.
+    obs_data : :py:class:`pandas.DataFrame`
+        A :py:class:`pandas.DataFrame` with sleep statistics from the observed scorer.
+        Rows are unique observations and columns are unique sleep statistics.
+        Shape, index, and columns must be identical to ``ref_data``.
+    ref_scorer : str
+        Name of the reference scorer.
+    obs_scorer : str
+        Name of the observed scorer.
+    agreement : float
+        Multiple of the standard deviation to plot agreement limits. The default is 1.96, which
+        corresponds to a 95% confidence interval if the differences are normally distributed.
+
+        .. note:: ``agreement`` gets adjusted for regression-modeled limits of agreement.
+    confidence : float
+        The percentage confidence interval for the confidence intervals that are applied to bias and
+        limits of agreement. The same confidence interval percentage is applied to both standard and
+        bootstrapped confidence intervals.
+    alpha : float
+        Alpha cutoff used for all assumption tests.
+    verbose : bool or str
+        Verbose level. Default (False) will only print warning and error messages. The logging
+        levels are 'debug', 'info', 'warning', 'error', and 'critical'. For most users the choice is
+        between 'info' (or ``verbose=True``) and warning (``verbose=False``).
+
+    Notes
+    -----
+    Sleep statistics that are identical between scorers are removed from analysis.
```

Comment:
Is this the standard way? Doesn't it over-emphasize discrepancies between scorers by keeping only differing observations?

Reply:
Shoot, nice catch! I only removed them early-on because of the some of the warnings/errors that would get raised when running some stats functions on identical scores. I will look into properly handling this.